src/modules/Elsa.Alterations.Core/Services/WorkflowInstanceFinder.cs

```diff
@@ -52,10 +52,24 @@ public async Task<IEnumerable<string>> FindAsync(AlterationWorkflowInstanceFilte
             else
                 workflowInstanceIds.IntersectWith(matchingWorkflowInstanceIds);
         }
+        
+        // Alterations must apply only to running workflows.
+        workflowInstanceIds = (await FilterRunningWorkflowInstancesAsync(workflowInstanceIds, cancellationToken)).ToHashSet();
 
         return workflowInstanceIds;
     }
 
+    private async Task<IEnumerable<string>> FilterRunningWorkflowInstancesAsync(IEnumerable<string> workflowInstanceIds, CancellationToken cancellationToken)
```

Comment:
Thank you. I wasn't sure I used the right naming. One the one hand, I was like: "I am filtering _out_ the other states". But then, what remains is the filtered _in_ results. In other words, from what perspective do you look at the term "filtered"? I will rename it then form the perspective of "to filter _out_ the not wanted states.

Reply:
So I asked ChatGPT about this situation. It actually makes the claim that in programming, we focus on _what to include_. Here's the explanation:

"In programming and data processing, the term "filter" is typically understood to mean the selection of elements that meet a certain condition, while excluding those that do not. The focus is on what is being included rather than what is being excluded. Therefore, if you are writing a method to select all objects with a status of Running, the name "FilterRunningObjects" more accurately describes the action of the method. This name implies that the method will return a collection of objects whose status is Running, effectively "filtering in" the objects of interest based on the specified criterion.

On the other hand, "FilterNonRunningObjects" suggests the opposite action: selecting objects that do not have the status of Running (i.e., objects that are either Idle or Completed). This would be misleading if the intended functionality is to select Running objects.

To summarize, you should go with "FilterRunningObjects" for clarity and adherence to common programming conventions regarding the term "filter." This naming convention makes it immediately clear to other developers what the method does: it filters the collection to include only those objects that are in the Running state."

Thanks again for the question, it prompted me to actually look into it.